src/analysis/vector/geometry_checker/qgsfeaturepool.h

```diff
@@ -46,6 +46,15 @@ class ANALYSIS_EXPORT QgsFeaturePool : public QgsFeatureSink SIP_ABSTRACT
      */
     bool getFeature( QgsFeatureId id, QgsFeature &feature );
 
+    /**
+     * Get features for the provided \a request. No features will be fetched
+     * from the cache and the request is sent directly to the underlying feature source.
+     * Results of the request are cached in the pool and the ids of all the features
+     * are returned. This can be used to warm the cache for a particular area of interest
+     * (bounding box) or other set of features.
+     */
+    QgsFeatureIds getFeatures( const QgsFeatureRequest &request ) SIP_SKIP;
```

Comment:
I think QgsFeaturePool should be marked as technology preview/non-stable api -- I don't think this whole class is ready yet to become part of the stable api contract.

Reply:
Actually this applies to all the geometry checker classes - can we mark them ALL as non-stable/technology preview?